Fix various typos

BioGeek opened the pull request:
This pull request fixes various typos found in docstrings and comments with [`misspellings`](https://pypi.python.org/pypi/misspellings).

It also found one typo that I didn't change because it was in a class attribute [`self.hsps_prelim_gapped_attemped`](https://github.com/biopython/biopython/blob/7575be181cae02b258fcc360464fb3c294a342fe/Bio/Blast/Record.py#L351):

```
./Bio/Blast/Record.py:319: attemped -> "attempted"
./Bio/Blast/Record.py:351: attemped -> "attempted"

```
----

I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ AND the _BSD 3-Clause License_.

I *am* happy to be thanked by name in the ``NEWS.rst`` and
``CONTRIB.rst`` files.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

peterjc commented on Bio/SeqIO/AbiIO.py:
```diff
@@ -354,7 +354,7 @@ def AbiIterator(handle, alphabet=None, trim=False):
         return
     if marker != b"ABIF":
-        raise IOError('File should start ABIF, not %r' % marker)
+        raise IOError('File should start with ABIF, not %r' % marker)
```
I think the original is OK here, and the improved wording does not justify changing the exception and the small chance of breaking someone's script it is matched on the message.

peterjc commented:
Most of these are comments or otherwise harmless, so the fixes are very welcome 👍 

Other than reverting one of the ABIF exception changes, this looks ready to merge.

Would you like to add yourself to the ``NEWS.rst`` and ``CONTRIB.rst`` file?

peterjc commented:
Also, unfortunately, it looks like you need to rebase to the current master (or merge) as GitHub says there are some merge conflicts. Are you happy doing that yourself?

BioGeek commented:
- I'm already mentioned in the `CONTRIB.rst` file
- I've added a small mention in the `NEWS.rst` file
- I've reverted the change to the ABIF exception

I'm reading up on how to do a rebase. I'll let you know if I succeed. Thanks!

BioGeek commented:
Oh crap, that reapplied some of the reverted changes from earlier. 
Sorry, I give up. I have no idea how to fix this mess.

peterjc commented:
No problem, leave it as is. I don't think I have time this afternoon, but I'll try to get to this later in the week

It looks like you ticked the box on GitHub giving me (and the other Biopython developers) permission to edit your branch, so I'll probably do that (with a ``git push BioGeek typos --force`` or similar).
